pyonfx/timestamps.py

```diff
@@ -0,0 +1,681 @@
+        ffprobe_output_dict = json.loads(ffprobe_output.stdout)
+
+        if not ffprobe_output_dict:
+            raise ValueError(f"The file {video_path} is not a video file.")
+
+        if len(ffprobe_output_dict["streams"]) == 0:
+            raise ValueError(f"The index {index} is not in the file {video_path}.")
+
+        if ffprobe_output_dict["streams"][0]["codec_type"] != "video":
+            raise ValueError(
+                f'The index {index} is not a video stream. It is an "{ffprobe_output_dict["streams"][0]["codec_type"]}" stream.'
+            )
+
+        if ffprobe_output_dict["format"]["format_name"] == "matroska,webm":
+            # We only do this check for .mkv file. See the note about mkv in the class documentation
+            time_base = Fraction(ffprobe_output_dict["streams"][0]["time_base"])
+            # 1/1000 represent 1 ms. If the time_base cannot divided by 1/1000, then it means that the timestamps aren't rounded
+            if time_base % (1 / 1000):
+                warnings.warn(
+                    "Your mkv file isn't perfectly rounded to ms. In this situation, you may prefer to use RoundingMethod.floor then RoundingMethod.ROUND.",
+                    UserWarning,
+                )
+
+        first_frame_time, last_frame_time, timestamps = get_timestamps(
+            ffprobe_output_dict["packets"]
+        )
+        fpms = Fraction(len(timestamps) - 1) / (last_frame_time - first_frame_time)
+
+        return cls(
+            rounding_method=rounding_method,
+            timestamps=timestamps,
+            normalize=normalize,
+            fpms=fpms,
+            last_frame_time=last_frame_time,
+        )
+
+    @staticmethod
+    def _from_mkvextract(
+        mkvextract_path: os.PathLike[str],
+        mkvmerge_path: os.PathLike[str],
+        video_path: os.PathLike[str],
+        index: int,
+        normalize: bool,
+        rounding_method: RoundingMethod,
+    ) -> Timestamps:
+        """Create timestamps based on the ``video_path`` provided.
+
+        Note:
+            This method requires the ``mkvextract`` and ``mkvmerge`` programs to be available.
+
+        Parameters:
+            mkvextract_path (os.PathLike[str]): The path to mkvextract.
+            mkvmerge_path (os.PathLike[str]): The path to mkvmerge.
+            video_path (os.PathLike[str]): A Video path.
+            index (int, optional): Stream index of the video.
+            normalize (bool, optional): If True, it will shift the timestamps to make them start from 0. If false, the option does nothing.
+            rounding_method (RoundingMethod, optional): A rounding method. See the comment in Timestamps description about floor vs round.
+        Returns:
+            An Timestamps instance.
+        """
+
+        cmd = [
+            mkvmerge_path,
+            video_path,
+            "-J",
+        ]
+        mkvmerge_output = subprocess.run(cmd, capture_output=True, text=True)
```

Comment:
Add ``--command-line-charset UTF-8 --output-charset UTF-8`` to the command. It is necessary when the user asks for a filepath that contains special characters that the locale encoding (which be queried with ``locale.getencoding()``) doesn't support.
```suggestion
        cmd = [
                mkvmerge_path,
                video_path,
                "-J",
                "--command-line-charset", "UTF-8",
                "--output-charset", "UTF-8",
        ]
        mkvmerge_output = subprocess.run(cmd, capture_output=True, text=True, encoding="utf-8")
```